include/zephyr/posix/posix_types.h

```diff
@@ -60,7 +60,8 @@ typedef struct k_sem sem_t;
 typedef uint32_t pthread_mutex_t;
 
 struct pthread_mutexattr {
-	int type;
+	unsigned char type: 2;
+	bool initialized: 1;
```

Comment:
Would it make sense to add rsvd bits to keep the size the same and not break ABI?

Reply:
What abi size specifically? I had seen these as 1 char in size elsewhere but they are usually passed around via pointers.

Reply:
In this case it was an int32, so maintaining that size here. Again not sure if it makes any sense in this case.

Reply:
I can't see too many scenarios where this would be abi-sensitive, but maybe I am missing some perspective. `pthread_mutexattr_t` should only be modified or queried via public accessor functions and should be passed around via pointer.

I'll leave this comment open though in case you feel it shouldn't be resolved yet or in case someone else feels that there is some additional perspective.

Reply:
okay.  I'm thinking my concern is a non-issue, but the thought being someone having a bit of code similar to `sizeof(struct pthread_mutexattr)` somewhere in the code (or some such size dependency like an offset into a table) that would be impacted here.  But your answer is saying that shouldn't be used that way ever, so... I'll mark this as resolved.  Thanks for considering the feedback!